src/core/Akka/Actor/ActorCell.cs

```diff
@@ -241,8 +240,12 @@ void IUntypedActorContext.Become(UntypedReceive receive, bool discardOld)
 
         private long NewUid()
         {
-            var auid = Interlocked.Increment(ref _uid);
-            return auid;
+            // Note that this uid is also used as hashCode in ActorRef, so be careful
+            // to not break hashing if you change the way uid is generated
+            var uid = ThreadLocalRandom.Current.Next();
```

Comment:
Replaced the incrementing counter with `ThreadLocalRandom` due to an easily reproducible issue where if a remote actor system restarts and creates its actors in the same order each time, the UIDs will always be the same. Observed this happening in some of the MNTR specifications I was debugging. Instead we'll use a random number identical to what the JVM does:https://github.com/akka/akka/blob/65a9e0a0f9372d2e2ecb945c8a9dae129a438f77/akka-actor/src/main/scala/akka/actor/ActorCell.scala#L350

Minus tail recursion.